PKG_CONFIG_USE_DEPENDENCY doesn't do the right thing: cmake insists on making
the distinction between LDFLAGS and target_link_libraries and
PKG_CONFIG_USE_DEPENDENCY puts everything in LDFLAGS. As a result, this fails
with binutils>2.22 (because LDFLAGS are put before .o files that require
libraries).

Another problem is that jrl-mathttols.pc is constructed so that it fools
cmake's pkg-config macro, so that libraries end up in JRL_MAL_LDFLAGS, not
_LIBRARIES... sigh :)

--- tests/CMakeLists.txt~	2011-12-07 15:35:56.000000000 +0100
+++ tests/CMakeLists.txt	2012-01-14 14:35:25.000000000 +0100
@@ -44,7 +44,10 @@
   TARGET_LINK_LIBRARIES(${NAME} ${Boost_LIBRARIES})
 
   # Search for jrl-mal.
-  PKG_CONFIG_USE_DEPENDENCY(${NAME} jrl-mal)
+  # PKG_CONFIG_USE_DEPENDENCY(${NAME} jrl-mal)
+  include_directories(${JRL_MAL_INCLUDE_DIRS})
+  link_directories(${JRL_MAL_LIBRARY_DIRS})
+  target_link_libraries(${NAME} ${JRL_MAL_LDFLAGS})
 ENDMACRO(DYNAMIC_GRAPH_TEST)
 
 
